src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -1175,6 +1156,9 @@ private static RemoteWalkContext CreateRemoteWalkContext(RestoreRequest request,
                 context.RemoteLibraryProviders.Add(provider);
             }
 
+            // Determine if the targets and props files should be written out.
+            context.IsMsBuildBased = request.ProjectStyle != ProjectStyle.DotnetCliTool;
```

Comment:
So `Unknown` and `ProjectJson` and `DotnetToolReference` are MsBuildBased?
```suggestion
            context.IsMsBuildBased = context.IsMsBuildBased && request.ProjectStyle != ProjectStyle.DotnetCliTool;
```

Reply:
If you look at the diff, this is not a functional change, it's just a timing change. 
The context property gets initialized at an earlier/more appropriate point.